### PR TITLE
Added grails BP

### PR DESCRIPTION
**Pull request by johanneswuerbach**
The grails BP is also preinstalled on Heroku https://devcenter.heroku.com/articles/buildpacks#default-buildpacks

### CODE REVIEW

**Comment by johanneswuerbach** [comment]
Double checked, but this test app and BP commit sha is working on heroku cedar-14.

Error here:
```
test-z2-app-grails
       -----> Grails app detected
tar: bin/util: Cannot open: Permission denied
tar: bin/java: Cannot open: Permission denied
tar: bin: Cannot utime: Operation not permitted
tar: jvm-buildpack-common.tar.gz: Cannot open: Permission denied
tar: readme.md: Cannot open: Permission denied
tar: Exiting with failure status due to previous errors
cat: /app/Procfile: No such file or directory
```

File: include/buildpack.bash [hunk 1]
```diff
@@ -94,6 +94,7 @@ buildpack-execute() {
 		exit 1
 	fi
 
+	cd "$build_path"
```
Comment (johanneswuerbach) [inline include/buildpack.bash]: Maybe we should execute this in another writeable directory. `heroku run ls -la` doesn't list the bin folder created by the grails bp.

**Comment by johanneswuerbach** [comment]
Rebased. @progrium it looks like having `bindata.go` inside the repo is causing a lot of conflicts. Couldn't it be removed as `make build` recreates the file anyway.

**Comment by johanneswuerbach** [comment]
@progrium anything blocking this? We would need this for deis to have full backwards compatibility.

File: include/procfile.bash [hunk 2]
```diff
@@ -33,7 +33,10 @@ procfile-types() {
 		local default_types
 		default_types="$(cat $app_path/.release | yaml-keys default_process_types | xargs echo)"
 		[[ "$default_types" ]] && \
-			echo "Default process types for $selected_name -> ${default_types// /, }"
+			echo "Default types for $selected_name -> ${default_types// /, }"
+			for type in $default_types; do
+				echo "$type: $(cat $app_path/.release | yaml-get default_process_types $type)" >> "$app_path/Procfile"
+			done
```
Comment (progrium) [inline include/procfile.bash]: is this indented correctly? it's always going run, is that desired? i suppose this is the problem with `&&` conditional statements.

File: include/procfile.bash [hunk 3]
```diff
@@ -33,7 +33,10 @@ procfile-types() {
 		local default_types
 		default_types="$(cat $app_path/.release | yaml-keys default_process_types | xargs echo)"
 		[[ "$default_types" ]] && \
-			echo "Default process types for $selected_name -> ${default_types// /, }"
+			echo "Default types for $selected_name -> ${default_types// /, }"
+			for type in $default_types; do
+				echo "$type: $(cat $app_path/.release | yaml-get default_process_types $type)" >> "$app_path/Procfile"
```
Comment (progrium) [inline include/procfile.bash]: also, is this going to always overwrite Procfile?

File: tests/apps/grails/.gitignore [hunk 4]
```diff
@@ -0,0 +1,17 @@
+*.iws
```
Comment (progrium) [inline tests/apps/grails/.gitignore]: we can drop this file

**Comment by progrium** [comment]
Just want to ensure that default types Procfile bit... it *looks* like it's always appending default types if they exist to Procfile, which I don't believe is desired when a Procfile exists, right?

File: tests/apps/grails/.gitignore (hunk 4 above)
Comment (johanneswuerbach) [inline tests/apps/grails/.gitignore]: I would prefer to keep this for testing whether this app still works in heroku

File: include/procfile.bash (hunk 2 above)
Comment (johanneswuerbach) [inline include/procfile.bash]: Sorry those lines are indented wrongly, but this should always run when we don't have a `Procfile` and a `.release`.

File: include/procfile.bash (hunk 3 above)
Comment (johanneswuerbach) [inline include/procfile.bash]: Yes, as commented above.

**Comment by johanneswuerbach** [comment]
No, because when a Procfile exists, those lines aren't executed, because you already returned. Not? https://github.com/johanneswuerbach/herokuish/blob/9c7050331131cdc1f4ede2628b28a598a661b6ba/include/procfile.bash#L26-L30

**Comment by progrium** [comment]
Cool, read it again in context and it all makes sense. Fix indent, leave gitignore and I'll merge. Thanks!

**Comment by progrium** [comment]
Open an issue for bindata. I'm open to it.

**Comment by johanneswuerbach** [comment]
Done

**Comment by progrium** [comment]
Oh, I was fine with gitignore, but ok. :)

**Pull request merged**